### PR TITLE
Disallow uncaught exceptions

### PR DESCRIPTION
**Pull request by jasonLaster**
We were allowing uncaught exceptions in our unit tests, which is a bad practice:

We now throw when we see it.

screen shot 2017-11-04 at 11 56 07 am

### CODE REVIEW

File: src/actions/tests/pause.spec.js [hunk 1]
```diff
@@ -2,7 +2,8 @@ import {
   actions,
   selectors,
   createStore,
-  getHistory
+  getHistory,
+  makeSource
```
Comment (jimblandy) [inline src/actions/tests/pause.spec.js]: Are these calls fixes for errors that had been previously being ignored? Usually one should submit unrelated fixes as a separate PR, and treat the harness fixes as blocked on that.

File: src/test/tests-setup.js [hunk 2]
```diff
@@ -28,4 +28,5 @@ setConfig(config);
 
 process.on("unhandledRejection", (reason, p) => {
   console.log("Unhandled Rejection at:", p, "reason:", reason);
+  throw reason;
```
Comment (jimblandy) [inline src/test/tests-setup.js]: A thousand times yes.

File: src/actions/tests/pause.spec.js (hunk 1 above)
Comment (jasonLaster) [reply src/actions/tests/pause.spec.js]: yep, this is a tad lazy... 

I don't think we gain anything in version control land by doing that though here
